rust/nextpnr/src/lib.rs

```diff
@@ -32,7 +32,7 @@ pub struct NetInfo {
 }
 
 impl NetInfo {
-    pub fn driver(&mut self) -> *mut PortRef {
+    pub fn driver(&mut self) -> &mut PortRef {
```

Comment:
hmm, I think this is one case where a null could exist - probably this should be converting the pointer to a ref in an option or something?